Add new github action to regenerate the datalab testkits on specific commit message

dtgreiner opened the pull request:
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Add a new github action that will regenerate the testkit fixpoints when a commit with "[gh:rebuild_fixpoints]" in the commit message.

https://github.com/open-path/Green-River/issues/5945

## Type of change
[//]: # 'remove options that are not relevant'
- [X] New feature (adds functionality)

## Checklist before requesting review
- [X] I have performed a self-review of my code
- [ ] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [X] My code follows the style guidelines of this project (rubocop)
- [ ] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

dtgreiner committed:
Add new github action to regenerate the datalab testkits on specific commit message

eanders commented on .github/workflows/test_kit_fixpoints.yml:
```diff
@@ -0,0 +1,204 @@
+          aws_region: us-east-1
+          flags: --recursive
+      - name: Remove existing fixpoint files
+        env:
+          AWS_ACCESS_KEY_ID: local_access_key
+          AWS_SECRET_ACCESS_KEY: local_secret_key
+          GROVER_NO_SANDBOX: true
+          CHROMIUM_PATH: /usr/bin/chromium-browser
+          MAX_FAILURES: 60
+          LOG_LEVEL: INFO
```
Probably don't need the ENV just for calling `rm` (though no harm either)

eanders commented on .github/workflows/test_kit_fixpoints.yml:
```diff
@@ -0,0 +1,204 @@
+          LOG_LEVEL: INFO
+        run: |
+          bundle exec rspec --fail-fast=$MAX_FAILURES --color -P "drivers/datalab_testkit/spec/empty_spec.rb"
+          OKTA_DOMAIN=localhost OKTA_CLIENT_ID=x OKTA_CLIENT_SECRET=x bundle exec rspec --fail-fast=$MAX_FAILURES --color -fd spec/requests/omniauth_spec.rb spec/requests/sessions_controller_spec.rb # these tests need okta enabled
```
```suggestion
```

eanders reviewed (approved):
@dtgreiner I think this looks generally good.  Can you push a commit that runs it, so we can see what happens? Probably best to make a duplicate of the existing files at a slightly different path and run it against that so we don't break/remove the existing files during test runs.

dtgreiner committed:
cleanup testkit action

dtgreiner committed:
hard-coding to an alternative test folder in the s3 bucket for testing

dtgreiner commented:
@eanders I've copied the testkit files in S3 to a new folder in the same bucket (test-kit-fixpoint-automation-test) and hardcoded the action to look there for copying and moving the fixpoints. the incoming commit should kick off the action.

dtgreiner committed:
[gh:rebuild_fixpoints]

dtgreiner committed:
remove step clearing existing fixpoint files, they do not exist yet

dtgreiner committed:
[gh:rebuild_fixpoints]

eanders commented:
@dtgreiner looks like it needs a little work, but great news, that the "test" ran to completion!  My biggest worry was that this would take too long.

dtgreiner commented:
@eanders it looks like it may be a permission issue. I don't have access to see the account making this call (github secrets - AWS_S3_ACCESS_KEY_ID) but I'm happy to reach out to devops to see if it can be updated, assuming it is the issue.

```
aws s3 cp ***test-kit-fixpoint-automation-test/current/ ***test-kit-fixpoint-automation-test/previous/ --recursive
copy failed: ***test-kit-fixpoint-automation-test/current/warehouse_app_test.sql to ***test-kit-fixpoint-automation-test/previous/warehouse_app_test.sql An error occurred (AccessDenied) when calling the CopyObject operation: Access Denied
copy failed: ***test-kit-fixpoint-automation-test/current/warehouse_test.sql to ***test-kit-fixpoint-automation-test/previous/warehouse_test.sql An error occurred (AccessDenied) when calling the GetObjectTagging operation: Access Denied
```

dtgreiner committed:
remove fixpoint generation for testing bucket copying [gh:rebuild_fixpoints]

dtgreiner committed:
include current date in path for previous fixpoints [gh:rebuild_fixpoints]

dtgreiner committed:
date as environment variable [gh:rebuild_fixpoints]

dtgreiner committed:
re-enable fixpoint generation for full test run [gh:rebuild_fixpoints]

dtgreiner commented:
@eanders the copy permissions appear to be resolved and I've kicked off a full test run to generate new fixpoints for the text directory in S3. If this runs successfully, I believe it should be ready to remove the hard-coded test directory. It should be running for a few hours to generate the fixpoints.

dtgreiner committed:
replace hardcoded test directory with github variables

dtgreiner commented:
@eanders the automated fixpoint generation process ran successfully. I've removed the hard-coded test directories and replaced them with the github variable directories. This PR should be ready to move forward.

eanders commented:
That's awesome!  Thank you!.  I'm re-targeting to 119 and merging.
